### PR TITLE
Add jittering when flushing data to disk

### PR DESCRIPTION
**Pull request by xichen2020**
cc @robskillington @martin-mao

### CODE REVIEW

**Comment by robskillington** [comment]
We might want to think about the implications of this, for background repair it's probably ok that some take a little longer than others to repair.  However for flushing to disk, delaying the beginning of that process means we extend the time where data is held purely in memory (with the exception of the commit log which is very expensive to bootstrap from).  I think we can get away with a very small jitter, perhaps say 2-5 minutes, but 30 minutes definitely seems to aggressive.

File: storage/fs.go [hunk 1]
```diff
@@ -55,11 +57,19 @@ func newFileSystemManager(database database) databaseFileSystemManager {
 	fm := newFlushManager(database)
 	cm := newCleanupManager(database, fm)
 
+	var jitter time.Duration
+	if maxJitter := opts.FileOpOptions().Jitter(); maxJitter > 0 {
+		nowFn := opts.ClockOptions().NowFn()
+		rand.Seed(nowFn().UnixNano())
```
Comment (robskillington) [inline storage/fs.go]: We should avoid setting the global rand seed here, we probably need to create a new rand source and then seed it.

Comment (xichen2020) [inline storage/fs.go]: sure can do

**Comment by xichen2020** [comment]
Agreed we want to flush as early as possible, although 2-5 minutes might not be not long enough to spread out disk flushing across all nodes given a reasonably large cluster. Can start with 5 minutes and go from there.

**Comment by robskillington** [comment]
Yeah, perhaps let's go with 5 minutes and I'll try to fix any underlying issues that cause degradation of perf during the actual flush.  

Can you perhaps fix the `dbRepairer` also setting a global `rand.Seed` to use it's own source?

**Pull request closed without merging**